### PR TITLE
Array operators

### PR DESCRIPTION
**Pull request by BastiaanOlij**
Add array operators to GD extensions

relies on https://github.com/godotengine/godot/pull/52739

### CODE REVIEW

File: test/src/example.cpp [hunk 1]
```diff
@@ -71,6 +71,14 @@ void Example::_bind_methods() {
 	BIND_CONSTANT(CONSTANT_WITHOUT_ENUM);
 }
 
+Example::Example() {
+	UtilityFunctions::print("Constructor.");
```
Comment (BastiaanOlij) [inline test/src/example.cpp:85]: Unrelated, added these for testing, but worth keeping in here I think.

**Comment by BastiaanOlij** [comment]
@vnen can you review and merge? looks like its working :)

File: test/src/example.cpp [hunk 2]
```diff
@@ -103,6 +112,16 @@ void Example::emit_custom_signal(const String &name, int value) {
 	emit_signal("custom_signal", name, value);
 }
 
+Array Example::test_array() const {
+	Array arr;
+
+	arr.resize(2);
+	arr[0] = Variant(1);
```
Comment (BastiaanOlij) [inline test/src/example.cpp:132]: @vnen so this test case currently crashes but seemingly not in relation to the array but the way the variants are accessed. 

Any chance you could try out my PR and see the crash for yourself? I haven't got a clue what is going on here.

**Comment by vnen** [comment]
I did look into this but I couldn't yet figure out why this is crashing. It definitely has a bogus Variant value, probably not properly constructed, but I can't tell where exactly it is.

**Pull request closed without merging**